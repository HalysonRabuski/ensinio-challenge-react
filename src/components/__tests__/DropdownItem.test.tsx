/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
// import '../../i18n'
// import renderer from 'react-test-renderer'
import DropdownItem from '../DropdownItem'

afterEach(() => {
  cleanup()
})

jest.mock('react-i18next', () => ({
  // this mock makes sure any components using the translate hook can use it without a warning being shown
  useTranslation: () => ({
    t: (str: any) => str,
    i18n: {
      changeLanguage: () => new Promise(() => {}),
    },
  }),
}))

test('should render card component and have test as text prop', async () => {
  render(<DropdownItem text="test" />)
  const DropdopwnElement = screen.getByTestId('dropdown-id')

  expect(DropdopwnElement).toBeInTheDocument()
  expect(DropdopwnElement).toHaveTextContent('test')
})

test('should render card component with an image and texts as children', () => {
  const DropdopwnElement = renderer
    .create(<DropdownItem text="test" />)
    .toJSON()
  expect(DropdopwnElement).toMatchSnapshot()
})
