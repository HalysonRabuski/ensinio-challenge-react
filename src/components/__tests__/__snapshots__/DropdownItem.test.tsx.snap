// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render card component with an image and texts as children 1`] = `
<div
  className="sc-bdnxRM cdUAXz"
  data-testid="dropdown-id"
>
  <div
    className="dropdown-item"
  >
    <a
      onClick={[Function]}
    >
      test
       
      <span>
         
      </span>
    </a>
  </div>
  <div
    className="dropdown-content"
  />
</div>
`;
